### PR TITLE
User variadic version and remove nil

### PR DESCRIPTION
**Pull request by vagababov**
Update duck/status to use new version of UnionMaps

/assign @vaikas mattmoor

### CODE REVIEW

File: apis/duck/v1/status_types.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ func (source *Status) ConvertTo(ctx context.Context, sink *Status, predicates ..
 	sink.ObservedGeneration = source.ObservedGeneration
 	if source.Annotations != nil {
 		// This will deep copy the map.
-		sink.Annotations = kmeta.UnionMaps(source.Annotations, nil)
+		sink.Annotations = kmeta.UnionMaps(source.Annotations)
```
Comment (vaikas) [inline apis/duck/v1/status_types.go:99]: nit, but maybe copy the comment about deep copying to the v1beta1 as well?

Comment (vagababov) [reply apis/duck/v1/status_types.go:99]: eh, sure.

**Comment by vagababov** [comment]
ping :)

**Pull request merged**